test/module-test.cc

```diff
@@ -55,11 +76,551 @@ TEST(module_test, detail_namespace) {
+}
+
+TEST(module_test, color) {
+  auto fg_check = fg(fmt::rgb(255, 200, 30));
+  auto bg_check = bg(fmt::color::dark_slate_gray) | fmt::emphasis::italic;
+  auto emphasis_check = fmt::emphasis::underline | fmt::emphasis::bold;
+  EXPECT_EQ("\x1B[30m42\x1B[0m",
+    fmt::format(fg(fmt::terminal_color::black), "{}", 42));
+}
+
+TEST(module_test, cstring_view) {
+  fmt::cstring_view nsv("fmt");
+  EXPECT_EQ("fmt", nsv.c_str());
+  fmt::wcstring_view wsv(L"fmt");
+  EXPECT_EQ(L"fmt", wsv.c_str());
+}
+
+TEST(module_test, buffered_file) {
+  EXPECT_TRUE(fmt::buffered_file{}.get() == nullptr);
+}
+
+TEST(module_test, move) {
+  fmt::ostream out = fmt::output_file("module-test", fmt::buffer_size = 1);
+  out.close();
+}
+
+struct custom_context {
+  using char_type = char;
+  using parse_context_type = fmt::format_parse_context;
+
+  bool called = false;
+
+  template <typename T> struct formatter_type {
+    auto parse(fmt::format_parse_context& ctx) -> decltype(ctx.begin()) {
+      return ctx.begin();
+    }
+
+    const char* format(const T&, custom_context& ctx) {
+      ctx.called = true;
+      return nullptr;
+    }
+  };
+
+  void advance_to(const char*) {}
+};
+
+struct test_struct {};
+
+namespace fmt {
+template <typename Char> struct formatter<test_struct, Char> {
+  auto parse(format_parse_context& ctx) -> decltype(ctx.begin()) {
+    return ctx.begin();
+  }
+
+  auto format(test_struct, format_context& ctx) -> decltype(ctx.out()) {
+    auto test = string_view("test");
+    return std::copy_n(test.data(), test.size(), ctx.out());
+  }
+};
+}
+
+TEST(arg_test, custom_context) {
+  auto ctx = custom_context();
+  auto parse_ctx = fmt::format_parse_context("");
+  fmt::basic_format_arg<custom_context> custom_arg;
+  EXPECT_TRUE(!custom_arg);
+
+  auto test = test_struct();
+  auto store = fmt::dynamic_format_arg_store<custom_context>();
+  store.push_back(test);
+}
+
+struct enabled_formatter {};
+struct disabled_formatter {};
+struct disabled_formatter_convertible {
+  operator int() const { return 42; }
+};
+
+namespace fmt {
+  template <> struct formatter<enabled_formatter> {
+  auto parse(format_parse_context& ctx) -> decltype(ctx.begin()) {
+    return ctx.begin();
+  }
+  auto format(enabled_formatter, format_context& ctx) -> decltype(ctx.out()) {
+    return ctx.out();
+  }
+};
+}
+
+TEST(core_test, has_formatter) {
+  using fmt::has_formatter;
+  using context = fmt::format_context;
+  static_assert(has_formatter<enabled_formatter, context>::value, "");
+  static_assert(!has_formatter<disabled_formatter, context>::value, "");
+  static_assert(!has_formatter<disabled_formatter_convertible, context>::value,
+                "");
+}
+
+TEST(core_test, is_formattable) {
+  static_assert(fmt::is_formattable<enabled_formatter>::value, "");
+  static_assert(!fmt::is_formattable<disabled_formatter>::value, "");
+  static_assert(fmt::is_formattable<disabled_formatter_convertible>::value, "");
+}
+
+struct convertible_to_int {
```

Comment:
Everything starting from here to the end of the file falls into the category of "functional testing", please remove. It is very unlikely that this functionality behaves differently between modular and non-modular build because it goes through the same argument handling paths as everything else.

Reply:
I've condensed it as much as possible to name-lookup.